src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/EngineGeneralTest.java

```diff
@@ -2293,6 +2294,96 @@ public void testCaseMultipleWhenElse() throws DatabaseEngineException {
+        engine.persist("TEST", entry().set("COL1", 2).set("COL5", "xpto").set("COL6", "xpto")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 3).set("COL5", "xpto").set("COL6", "xpto")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 4).set("COL5", "teste").set("COL6", "teste")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 5).set("COL5", "pomme de terre").set("COL6", "pomme de terre")
+                                      .build());
+
+        final Query query =
+                select(
+                        concat(k(""),
+                               column("COL5"),
+                               column("COL6")).alias("concat"))
+                        .from(table("TEST"));
+
+        final List<Map<String, ResultColumn>> result = engine.query(query);
+
+        assertEquals("testeteste", result.get(0).get("concat").toString());
+        assertEquals("xptoxpto", result.get(1).get("concat").toString());
+        assertEquals("xptoxpto", result.get(2).get("concat").toString());
+        assertEquals("testeteste", result.get(3).get("concat").toString());
+        assertEquals("pomme de terrepomme de terre", result.get(4).get("concat").toString());
+    }
+
+    @Test
+    public void testConcatColumn() throws DatabaseEngineException {
+        test6Columns();
+        engine.persist("TEST", entry().set("COL1", 1).set("COL5", "teste").set("COL6", "teste")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 2).set("COL5", "xpto").set("COL6", "xpto")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 3).set("COL5", "xpto").set("COL6", "xpto")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 4).set("COL5", "teste").set("COL6", "teste")
+                                      .build());
+        engine.persist("TEST", entry().set("COL1", 5).set("COL5", "pomme de terre").set("COL6", "pomme de terre")
+                                      .build());
+
+        final Query query =
+                select(
+                        concat(column("COL5"),
+                               column("COL5"),
+                               column("COL6")).alias("concat"))
+                        .from(table("TEST"));
+
+        final List<Map<String, ResultColumn>> result = engine.query(query);
+
+        assertEquals("testetesteteste", result.get(0).get("concat").toString());
+        assertEquals("xptoxptoxpto", result.get(1).get("concat").toString());
+        assertEquals("xptoxptoxpto", result.get(2).get("concat").toString());
+        assertEquals("testetesteteste", result.get(3).get("concat").toString());
+        assertEquals("pomme de terrepomme de terrepomme de terre", result.get(4).get("concat").toString());
+    }
```

Comment:
All tests seem very similar can you just make a single function and pass the differences as an argument?

Reply:
While I agree they are similar the assertions are different. WDYT of refactoring everything except the assertions?